cloudflare/resource_cloudflare_api_token.go

```diff
@@ -0,0 +1,248 @@
+	policies := d.Get("policy").([]interface{})
+	var cfPolicies []cloudflare.APITokenPolicies
+
+	for _, p := range policies {
+		policy := p.(map[string]interface{})
+
+		permissionGroups := expandInterfaceToStringList(policy["permission_groups"])
+		var cfPermissionGroups []cloudflare.APITokenPermissionGroups
+		for _, pg := range permissionGroups {
+			cfPermissionGroups = append(cfPermissionGroups, cloudflare.APITokenPermissionGroups{
+				ID: pg,
+			})
+		}
+
+		cfResources := map[string]interface{}{}
+		for k, v := range policy["resources"].(map[string]interface{}) {
+			// value can be object or just a string ("*"), try to convert it to map
+			obj := map[string]string{}
+			if err := json.Unmarshal([]byte(v.(string)), &obj); err == nil {
+				cfResources[k] = obj
+			} else {
+				cfResources[k] = v
+			}
+		}
+
+		cfPolicies = append(cfPolicies, cloudflare.APITokenPolicies{
+			Effect:           policy["effect"].(string),
+			Resources:        cfResources,
+			PermissionGroups: cfPermissionGroups,
+		})
+	}
+
+	return cfPolicies
+}
+
+func resourceCloudflareApiTokenRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*cloudflare.API)
+	tokenID := d.Id()
+
+	t, err := client.GetAPIToken(tokenID)
+
+	log.Printf("[DEBUG] Cloudflare APIToken: %+v", t)
+	log.Printf("[DEBUG] Cloudflare APIToken error: %#v", err)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "HTTP status 404") {
+			log.Printf("[INFO] Cloudflare ApiToken %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error finding Cloudflare ApiToken %q: %s", d.Id(), err)
+	}
+
+	policies := []map[string]interface{}{}
+
+	for _, p := range t.Policies {
+		permissionGroups := []string{}
+		for _, v := range p.PermissionGroups {
+			permissionGroups = append(permissionGroups, v.ID)
+		}
+
+		policies = append(policies, map[string]interface{}{
+			"resources":         p.Resources,
+			"permission_groups": permissionGroups,
+			"effect":            p.Effect,
+		})
+	}
+
+	d.Set("name", t.Name)
+	d.Set("policies", policies)
+	d.Set("status", t.Status)
+	d.Set("issued_on", t.IssuedOn.Format(time.RFC3339Nano))
+	d.Set("modified_on", t.ModifiedOn.Format(time.RFC3339Nano))
+	d.Set("request_ip_in", t.Condition.RequestIP.In)
+	d.Set("request_ip_not_in", t.Condition.RequestIP.NotIn)
+
+	return nil
+}
+
+func resourceCloudflareApiTokenUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	name := d.Get("name").(string)
+	tokenID := d.Id()
+
+	t := cloudflare.APIToken{
+		Name:      d.Get("name").(string),
+		Policies:  resourceDataToApiTokenPolices(d),
+		Condition: resourceDataToApiTokenCondition(d),
+	}
+
+	log.Printf("[INFO] Updating Cloudflare ApiToken: name %s", name)
+
+	t, err := client.UpdateAPIToken(tokenID, t)
+	if err != nil {
+		return fmt.Errorf("Error updating Cloudflare ApiToken %q: %s", name, err)
+	}
+
+	d.Set("modified_on", t.ModifiedOn.Format(time.RFC3339Nano))
```

Comment:
Instead of setting explicitly here, the method should call the `Read` method and it will sync all the values with the remote state instead of just the modified value. The [Terraform documentation](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#update-state-after-modification) has a good write up on this.